src/lib.rs

```diff
@@ -411,11 +448,35 @@ impl Datadog {
         tags: impl IntoIterator<Item = String>,
     ) {
         if self.is_reporting_enabled {
-            self.client.event(
-                metric.as_ref(),
-                text.as_ref(),
-                tags.into_iter().collect::<Vec<String>>(),
-            );
+            let tags = tags.into_iter().collect::<Vec<String>>();
+            let metric = metric.as_ref();
+            self.track_tags(metric, &tags);
+            self.client.event(metric, text.as_ref(), tags);
+        }
+    }
+
+    fn track_tags<'a>(&self, metric: impl AsRef<str>, tags: impl IntoIterator<Item = &'a String>) {
+        let hashes = tags
+            .into_iter()
+            .map(|t| {
+                let mut hasher = DefaultHasher::new();
+                hasher.write(metric.as_ref().as_bytes());
+                hasher.write(t.as_bytes());
+                hasher.finish()
+            })
+            .collect::<Vec<u64>>();
+        let mut seen_tags = self.tag_tracker.seen_tags.lock().unwrap();
+        seen_tags.extend(hashes);
+        let seen_tag_count = seen_tags.len();
+        drop(seen_tags);
+        if seen_tag_count >= self.tag_tracker.warn_threshold
+            && self
+                .tag_tracker
+                .sent_event
+                .compare_exchange(false, true, Ordering::SeqCst, Ordering::SeqCst)
+                .is_ok()
+        {
+            self.do_event(String::from(TAG_LIMIT_EXCEEDED), "Exceeded tag limit", vec![])
```

Comment:
We currently don't have tracing or log as a dependency but I think it would be nice to emit a log as well, especially for development

Reply:
I could go either way on this, one the one hand I don't want to add the dep but on the other you're right it would be nice to record this in two places potentially, and a big WARN in someones logs could help with them notice

Anyone else got a preference?